From e4aea4fc6522569e1851835148672c3a0ac12e80 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 1 Dec 2017 13:09:30 +0000
Subject: [PATCH] builder: Require `/proc` mount for new eopkg mechanisms

We don't actually _need_ dbus anymore, but we'll keep it around to
support legacy images that need updating. We need `/proc` mounted so
that `usysconf` can safely run within the chroot and perform any last
configuration tasks, as eopkg is no longer directly responsible for this.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/builder/update.go | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/builder/update.go b/src/builder/update.go
index b2fa48d..7d190b1 100644
--- a/src/builder/update.go
+++ b/src/builder/update.go
@@ -20,6 +20,7 @@ import (
 	log "github.com/Sirupsen/logrus"
 	"github.com/solus-project/libosdev/disk"
 	"os"
+	"path/filepath"
 )
 
 func (b *BackingImage) updatePackages(notif PidNotifier, pkgManager *EopkgManager) error {
@@ -111,6 +112,19 @@ func (b *BackingImage) Update(notif PidNotifier, pkgManager *EopkgManager) error
 		return err
 	}
 
+	procPoint := filepath.Join(b.RootDir, "proc")
+
+	// Bring up proc
+	log.WithFields(log.Fields{
+		"vfs": "/proc",
+	}).Debug("Mounting vfs")
+	if err := mountMan.Mount("proc", procPoint, "proc", "nosuid", "noexec"); err != nil {
+		log.WithFields(log.Fields{
+			"error": err,
+		}).Error("Failed to mount /proc")
+		return err
+	}
+
 	// Hand over to package management to do the updates
 	if err := b.updatePackages(notif, pkgManager); err != nil {
 		return err
-- 
2.15.1

